### PR TITLE
Add functional tests for Paying For College pages including search features

### PR DESCRIPTION
**Pull request by cwdavies**
This change adds integration tests for the four Paying For College pages and autocomplete search features.

## How to test this PR

Run `yarn run cypress open` and select the following cypress integration tests for the 4 paying for college pages:

1. choose-a-student-loan.js
2. manage-your-college-money.js
3. repay-student-debt.js
4. your-financial-path-to-graduation.js

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)

### CODE REVIEW

File: test/cypress/integration/pages/pfc-school-search.js [hunk 1]
```diff
@@ -5,14 +5,14 @@ const page = new PfcSchoolSearch();
 describe( 'Paying For College Search', () => {
   it( 'should return results', () => {
     page.open();
-    page.click( 'Get started' )
+    page.click( 'Get started' );
     page.enter( 'American' );
     page.searchResults().should( 'be.visible' );
-} );
+  } );
```
Comment (texasdude40) [inline test/cypress/integration/pages/pfc-school-search.js:11]: This is fraud this is David English this is my go affidavit

File: test/cypress/integration/pages/repay-student-debt.js [hunk 2]
```diff
@@ -7,25 +7,25 @@ describe( 'Paying for College', () => {
     it( 'should display Direct debit and extra payments', () => {
       page.open();
       page.click( 'Federal' );
-      page.selectQuestion ( '2' );
+      page.selectQuestion( '2' );
```
Comment (texasdude40) [inline test/cypress/integration/pages/repay-student-debt.js:10]: I don’t believe the government curtain hurt me I would hope not because I love my country

**Review by willbarton — APPROVED** [review]

File: test/cypress/pages/paying-for-college/school-search.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    cy.visit( `/paying-for-college2/understanding-your-financial-aid-offer/api/program/${ name }/` );
+  }
+
+  openSchool( id ) {
+    cy.visit( `/paying-for-college2/understanding-your-financial-aid-offer/api/school/${ id }.json` );
+  }
+
+  school( term ) {
+    cy.visit( `/paying-for-college2/understanding-your-financial-aid-offer/api/search-schools.json?q=${ term }` );
+  }
```
Comment (Scotchester) [inline test/cypress/pages/paying-for-college/school-search.js:25]: Is there a plan to use these currently-unused API visit methods in the near future? If not, let's remove them.

**Review by Scotchester — CHANGES_REQUESTED** [review]
This is a great start! I have one specific comment and one high-level comment.

First, would you mind moving the school search test into the `your-financial-path-to-graduation` tests? Since it's on the same page, I think they should be kept together.

More generally, I'd like to see more assertions. While having code that steps through a lot of the functionality is helpful to raise some major flags (e.g., the page starts 404ing or Cypress can't find an element that goes missing), I think we still want to confirm by assertions that we are seeing the resulting behavior we're expecting to see.

For example, on Your Financial Path, on the "Make a plan" step, you could assert that the input given earlier always results in having $579,234 of excess funding.

File: test/cypress/integration/pages/your-financial-path-to-graduation.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+const page = new PfcFinancialPathToGraduation();
+
+describe( 'Paying For College', () => {
+  describe( 'Your Financial Path To Graduation', () => {
+    it( 'should display total costs', () => {
+      page.open();
+      page.click( 'Get started' );
+      page.setText( 'search__school-input', 'Harvard' );
+      page.selectProgram( 'type', 'bachelors' );
```
Comment (higs4281) [inline test/cypress/integration/pages/your-financial-path-to-graduation.js:11]: An idea for more assertions, as mentioned by @Scotchester: 
If you choose 'type', 'graduate', there should be options in the `program-select` selection box.

**Review by higs4281 — APPROVED** [review]
This is a great addition to our testing suite. One minor suggestion for adding an assertion.

File: test/cypress/pages/paying-for-college/school-search.js (hunk 3 above)
Comment (cwdavies) [reply test/cypress/pages/paying-for-college/school-search.js]: I was not sure if there were any existing paying-for-college API tests. If you know that the API is tested by unit tests then these API visit methods can be removed.

**Comment by cwdavies** [comment]
I moved the school search test into the your-financial-path-to-graduation tests since they are on the same page.

File: test/cypress/integration/pages/your-financial-path-to-graduation.js (hunk 4 above)
Comment (cwdavies) [reply test/cypress/integration/pages/your-financial-path-to-graduation.js:87]: I added assertions using `page.getText` to verify total costs.
Also, I added tests for all program types: certificate, associates, graduate, as well as bachelors.

**Review by Scotchester — CHANGES_REQUESTED** [review]
Thanks for the updates, Craig. One more request: Would you please move all of the assertions to the files within the `integration/` directory? That will make it easier to see the connection between the steps taken in the test and what the expected results should be.

**Comment by cwdavies** [comment]
> Thanks for the updates, Craig. One more request: Would you please move all of the assertions to the files within the `integration/` directory? That will make it easier to see the connection between the steps taken in the test and what the expected results should be.

@Scotchester Please would you provide an example for an assertion in the `integration/` directory to clarify.
For example in `your-financial-path-to-graduation` most of the steps are to `set text` for each college cost.
There is now a check to `get text` which verifies that `total_costs` has the correct dollar amount which is an expected result.

File: test/cypress/integration/pages/choose-a-student-loan.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+      cy.get( 'div' ).should( 'contain', 'I have to borrow money for school. What are my options?' );
+      cy.get( '#answer1' ).should( 'contain', 'If you have to take out student loans' );
+      page.openLoanOption( 'o2' );
+      cy.get( 'div' ).should( 'contain', "What if my grants and federal loans don't cover the cost of attendance?" );
+      cy.get( '#answer2' ).should( 'contain', 'If your grants and federal loans are not enough to cover the cost of your education' );
+      page.openLoanOption( 'o3' );
+      cy.get( 'div' ).should( 'contain', 'What should I consider when shopping for a private loan?' );
+      cy.get( '#answer3' ).should( 'contain', 'First, make sure you need a private student loan' );
+    } );
+    it( 'should display options for each student loan', () => {
```
Comment (Scotchester) [inline test/cypress/integration/pages/choose-a-student-loan.js:18]:
```suggestion
    it( 'should display expandable comparison tables', () => {
```

File: test/cypress/integration/pages/choose-a-student-loan.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+      cy.get( '.compare-table' ).should( 'not.be.visible' );
+      page.clickOption( 'Federal Loan Options' );
+      cy.get( '#FederalLoanOptions' ).should( 'be.visible' );
+      page.clickOption( 'Private Loan Options' );
+      cy.get( '#PrivateLoanOptions' ).should( 'be.visible' );
+      cy.get( '#FederalLoanOptions' ).should( 'not.be.visible' );
+      page.clickOption( 'Private Loan Options' );
+      cy.get( '#PrivateLoanOptions' ).should( 'not.be.visible' );
+    } );
+    it( 'should display answers to each student loan option', () => {
```
Comment (Scotchester) [inline test/cypress/integration/pages/choose-a-student-loan.js:32]:
```suggestion
    it( 'should display answers to each "What are the options?" question', () => {
```

File: test/cypress/integration/pages/choose-a-student-loan.js [hunk 7]
```diff
@@ -1,39 +1,55 @@
...
+      page.openLoanOption( 'o1' );
+      cy.get( 'div' ).should( 'contain', 'I have to borrow money for school. What are my options?' );
+      cy.get( '#answer1' ).should( 'contain', 'If you have to take out student loans' );
+      page.openLoanOption( 'o2' );
+      cy.get( 'div' ).should( 'contain', "What if my grants and federal loans don't cover the cost of attendance?" );
+      cy.get( '#answer2' ).should( 'contain', 'If your grants and federal loans are not enough to cover the cost of your education' );
+      page.openLoanOption( 'o3' );
+      cy.get( 'div' ).should( 'contain', 'What should I consider when shopping for a private loan?' );
+      cy.get( '#answer3' ).should( 'contain', 'First, make sure you need a private student loan' );
+    } );
```
[the remark this answers is not in the document]
Comment (Scotchester) [reply test/cypress/integration/pages/choose-a-student-loan.js:17]: Also, I think maybe that `cy.visit( '/paying-for-college/choose-a-student-loan/' )` could be moved outside of the `it`, just inside the `describe`, and then it doesn't need to be called again for each `it`.

File: test/cypress/integration/pages/choose-a-student-loan.js [hunk 8]
```diff
@@ -0,0 +1,56 @@
+import { PayingForCollege } from '../../components/paying-for-college/';
+
+const page = new PayingForCollege();
+
+describe( 'Paying for College', () => {
```
Comment (Scotchester) [inline test/cypress/integration/pages/choose-a-student-loan.js:5]: I think this outer `describe` is unnecessary.

File: test/cypress/components/paying-for-college.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+    cy.visit( `/paying-for-college/choose-a-student-loan/#${ name }` );
+  }
+
+  openMoneyOption( name ) {
+    cy.visit( `/paying-for-college/manage-your-college-money/#${ name }` );
+  }
+
+  clickOption( name ) {
+    cy.get( 'span' ).contains( name ).click();
+  }
```
Comment (Scotchester) [inline test/cypress/components/paying-for-college.js:13]:
```suggestion
  clickExpandable( name ) {
    cy.get( 'span' ).contains( name ).click();
  }
```

File: test/cypress/components/paying-for-college.js [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+  selectOption( name ) {
+    cy.get( '.bubble-top-text' ).contains( name ).click();
+  }
```
Comment (Scotchester) [inline test/cypress/components/paying-for-college.js:17]: Let's call these bubbles, to match the class nomenclature.

```suggestion
  clickBubble( name ) {
    cy.get( '.bubble-top-text' ).contains( name ).click();
  }
```

File: test/cypress/components/paying-for-college.js [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+
+  closeOption( name ) {
+    cy.get( '.bubble-transparent-answer' ).find( 'p' ).contains( name ).click();
+  }
```
Comment (Scotchester) [inline test/cypress/components/paying-for-college.js:21]: This is no longer used.

```suggestion
```

File: test/cypress/components/paying-for-college.js [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+    } );
+  }
+
+  closeFirstOption() {
+    cy.get( '.btn-close' ).first().click();
+  }
+
+  closeLastOption() {
+    cy.get( '.btn-close' ).last().click();
+  }
```
Comment (Scotchester) [inline test/cypress/components/paying-for-college.js:35]: Let's carry on the Bubble naming and make the selectors a little more specific.

```suggestion
  closeAllBubbles() {
    cy.get( '.bubble-space .btn-close' ).each( el => {
      cy.wrap( el ).click( { force: true } );
    } );
  }

  closeFirstBubble() {
    cy.get( '.bubble-space .btn-close' ).first().click();
  }

  closeLastBubble() {
    cy.get( '.bubble-space .btn-close' ).last().click();
  }
```

File: test/cypress/integration/pages/choose-a-student-loan.js [hunk 13]
```diff
@@ -0,0 +1,56 @@
...
+        .and( 'contain', 'Congress has the authority to change federal student loan rates' );
+      page.selectOption( 'Should I use a credit card to cover my education costs?' );
+      cy.get( '.bubble-transparent-answer' ).should( 'be.visible' )
+        .and( 'contain', "Don't replace student loan debt with credit card debt" );
+      page.selectOption( "What if I can't repay my private student loan?" );
+      cy.get( '.bubble-transparent-answer' ).should( 'be.visible' )
+        .and( 'contain', 'Contact the company that services your student loan immediately' );
+      page.closeLastOption();
+      page.closeAllOptions();
+      cy.get( '.bubble-transparent-answer' ).should( 'not.be.visible' );
```
Comment (Scotchester) [inline test/cypress/integration/pages/choose-a-student-loan.js:53]: Convert all these to the Bubble naming as suggested to be modified in the component file above.

Also, I'm not sure that the `.and( 'contain'…`s are useful. The visibility assertion should be enough.

**Review by Scotchester — COMMENTED** [review]
Hey Craig, good progress! A few more comments and suggestions on the Choose a Student Loan tests, which should also be applied to the Manage Your College Money tests.

File: test/cypress/components/paying-for-college.js (hunk 11 above)
Comment (cwdavies) [reply test/cypress/components/paying-for-college.js]: Removed

File: test/cypress/components/paying-for-college.js (hunk 12 above)
Comment (cwdavies) [reply test/cypress/components/paying-for-college.js]: Changed

File: test/cypress/integration/pages/choose-a-student-loan.js (hunk 13 above)
Comment (cwdavies) [reply test/cypress/integration/pages/choose-a-student-loan.js]: I feel strongly that the `.and( 'contain` are useful.
It is important to check that a bubble answer is visible and contains text.
However if the text changes the test should fail.

File: test/cypress/components/paying-for-college.js (hunk 10 above)
Comment (cwdavies) [reply test/cypress/components/paying-for-college.js]: Changed Option to Bubble

File: test/cypress/components/paying-for-college.js (hunk 9 above)
Comment (cwdavies) [reply test/cypress/components/paying-for-college.js]: Changed Option to Bubble

File: test/cypress/integration/pages/choose-a-student-loan.js (hunk 8 above)
Comment (cwdavies) [reply test/cypress/integration/pages/choose-a-student-loan.js:5]: To be consistent with other tests I am leaving the outer describe and making no changes to this format.

File: test/cypress/integration/pages/choose-a-student-loan.js (hunk 6 above)
Comment (cwdavies) [reply test/cypress/integration/pages/choose-a-student-loan.js]: Done

File: test/cypress/integration/pages/choose-a-student-loan.js (hunk 5 above)
Comment (cwdavies) [reply test/cypress/integration/pages/choose-a-student-loan.js]: Done

File: test/cypress/integration/pages/choose-a-student-loan.js [hunk 14]
```diff
@@ -0,0 +1,56 @@
+import { PayingForCollege } from '../../components/paying-for-college/';
+
+const page = new PayingForCollege();
+
+describe( 'Paying for College', () => {
+  describe( 'Choose a student loan', () => {
+    it( 'should display answers to each student loan question', () => {
```
[the remark this answers is not in the document]
Comment (cwdavies) [reply test/cypress/integration/pages/choose-a-student-loan.js]: Done

File: test/cypress/integration/pages/choose-a-student-loan.js (hunk 7 above)
[the remark this answers is not in the document]
Comment (cwdavies) [reply test/cypress/integration/pages/choose-a-student-loan.js]: 1. Changed each 'div' to '#_o1', '#_o2', and '#_o3' to make the assertion run faster.
2. Created `page.openLoan` and `page.openMoney` that has a `cy.visit` for the url with the option number
3. Used `cy.get( '#_o1')`, `cy.get( '#_o2')`, and `cy.get( '#_o3')`
4. Verified `should( 'be.visible1 )` is being used in assertions

File: test/cypress/integration/pages/choose-a-student-loan.js (hunk 8 above)
Comment (Scotchester) [reply test/cypress/integration/pages/choose-a-student-loan.js:5]: Yeah, sorry about this one. Having looked at more of our current test suite, I see this pattern in use on some of them. I don't find it helpful, but it's fine until we can get better guidelines in place to ensure consistency.

File: test/cypress/integration/pages/choose-a-student-loan.js [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+const page = new PayingForCollege();
+
+describe( 'Paying for College', () => {
+  beforeEach( () => {
+    page.openLoan();
+  } );
+  describe( 'Choose a student loan', () => {
+    it( 'should display answers to each question', () => {
+      cy.get( '#_o1' ).click();
+      cy.get( '#_o1' ).should( 'contain', 'I have to borrow money for school. What are my options?' );
```
Comment (Scotchester) [inline test/cypress/integration/pages/choose-a-student-loan.js:12]: This is always visible on page load will never be false unless the text changes, and Cypress tests shouldn't generally be in the business of testing for changing content.

```suggestion
```

File: test/cypress/integration/pages/choose-a-student-loan.js [hunk 16]
```diff
@@ -0,0 +1,70 @@
...
+      cy.get( '#answer1' ).should( 'be.visible' )
+        .and( 'contain', 'If you have to take out student loans' );
+
+      cy.get( '#_o2' ).click();
+      cy.get( '#_o2' ).should( 'contain', "What if my grants and federal loans don't cover the cost of attendance?" );
```
Comment (Scotchester) [inline test/cypress/integration/pages/choose-a-student-loan.js:17]: This is always visible on page load will never be false unless the text changes, and Cypress tests shouldn't generally be in the business of testing for changing content.

```suggestion
```

File: test/cypress/integration/pages/choose-a-student-loan.js [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+      cy.get( '#answer2' ).should( 'be.visible' )
+        .and( 'contain', 'If your grants and federal loans are not enough to cover the cost of your education' );
+
+      cy.get( '#_o3' ).click();
+      cy.get( '#_o3' ).should( 'contain', 'What should I consider when shopping for a private loan?' );
```
Comment (Scotchester) [inline test/cypress/integration/pages/choose-a-student-loan.js:22]: This is always visible on page load will never be false unless the text changes, and Cypress tests shouldn't generally be in the business of testing for changing content.

```suggestion
```

File: test/cypress/integration/pages/manage-your-college-money.js [hunk 18]
```diff
@@ -0,0 +1,55 @@
...
+const page = new PayingForCollege();
+
+describe( 'Paying for College', () => {
+  beforeEach( () => {
+    page.openMoney();
+  } );
+  describe( 'Manage your college money', () => {
+    it( 'should display answers to each question', () => {
+      cy.get( '#_o1' ).click();
+      cy.get( '#_o1' ).should( 'contain', 'When should I get a bank account?' );
```
Comment (Scotchester) [inline test/cypress/integration/pages/manage-your-college-money.js:12]: This is always visible on page load will never be false unless the text changes, and Cypress tests shouldn't generally be in the business of testing for changing content.

```suggestion
```

File: test/cypress/integration/pages/manage-your-college-money.js [hunk 19]
```diff
@@ -0,0 +1,55 @@
...
+      cy.get( '#answer1' ).should( 'be.visible' )
+        .and( 'contain', 'Choose an account as soon as possible' );
+
+      cy.get( '#_o2' ).click();
+      cy.get( '#_o2' ).should( 'be.visible' )
+        .and( 'contain', 'How do I avoid paying unexpected fees?' );
+      cy.get( '#answer2' ).should( 'contain', 'very few accounts charge no fees at all' );
```
Comment (Scotchester) [inline test/cypress/integration/pages/manage-your-college-money.js:19]: Looks like you inserted the be-visible test on the wrong item here, and as above we don't need to test what's in `#_o2`.

```suggestion
      cy.get( '#answer2' ).should( 'be.visible' )
        .and( 'contain', 'very few accounts charge no fees at all' );
```

File: test/cypress/integration/pages/manage-your-college-money.js [hunk 20]
```diff
@@ -0,0 +1,55 @@
...
+
+      cy.get( '#_o3' ).click();
+      cy.get( '#_o3' ).should( 'be.visible' )
+        .and( 'contain', 'Do I have to get an account with the bank at my school?' );
+      cy.get( '#answer3' ).should( 'contain', 'Schools cannot require you to use their bank' );
```
Comment (Scotchester) [inline test/cypress/integration/pages/manage-your-college-money.js:24]: Looks like you inserted the be-visible test on the wrong item here, and as above we don't need to test what's in `#_o3`.

```suggestion
      cy.get( '#answer3' ).should( 'be.visible' )
        .and( 'contain', 'Schools cannot require you to use their bank' );
```

**Review by Scotchester — CHANGES_REQUESTED** [review]
This is very close now, but a few more suggestions.

> However if the text changes the test should fail.

I have to respectfully disagree with you on this one. Some of the suggested changes below get at this same question. Let me try and restate my opinion and see if it changes your mind at all :) If not, perhaps we should take the discussion to others on the team.

The two primary goals of the Cypress tests are to:
1. Watch out for user-facing regressions that we might otherwise not notice
2. Test user interactions that are difficult or impossible to test with unit tests

Testing to see that content that is hard-coded into a template has not changed does not contribute to either of these goals. Hard-coded content will never* change unless those lines of code are edited, and if someone does that, they're always** doing it intentionally and would simply be annoyed by the resulting Cypress test failure.

_* Unless that hard-coded content is later manipulated by JavaScript. In the case of these interactions, that is not the case. It is simply made visible or hidden._
_** Unless they made a typo while editing that file. Possible, but unlikely, and hopefully caught in code review._

The ideal use case for should-contain assertions is something like the cost calculations on Your Financial Path. We want to ensure that given a certain set of input values, the resulting total that is output on the page is calculated consistently. But for the three types of interactions we have on these Choose a Student Loan and Manage Your College Money pages, all we really need to confirm is that the selected content becomes visible (or goes away).

Let me know what you think!

File: test/cypress/integration/pages/manage-your-college-money.js (hunk 19 above)
Comment (cwdavies) [reply test/cypress/integration/pages/manage-your-college-money.js]: Done

File: test/cypress/integration/pages/manage-your-college-money.js (hunk 18 above)
Comment (cwdavies) [reply test/cypress/integration/pages/manage-your-college-money.js]: The content changes when you click different answers to a question which is what is being tested.
The test should fail if the content does not change for the different questions.

File: test/cypress/integration/pages/choose-a-student-loan.js (hunk 17 above)
Comment (cwdavies) [reply test/cypress/integration/pages/choose-a-student-loan.js]: The content changes when you click different answers to a question which is what is being tested.
The test should fail if the content does not change for the different questions.

File: test/cypress/integration/pages/choose-a-student-loan.js (hunk 16 above)
Comment (cwdavies) [reply test/cypress/integration/pages/choose-a-student-loan.js]: The content changes when you click different answers to a question which is what is being tested.
The test should fail if the content does not change for the different questions.

File: test/cypress/integration/pages/choose-a-student-loan.js (hunk 15 above)
Comment (cwdavies) [reply test/cypress/integration/pages/choose-a-student-loan.js]: The content changes when you click different answers to a question which is what is being tested.
The test should fail if the content does not change for the different questions.

**Comment by cwdavies** [comment]
> This is very close now, but a few more suggestions.
> 
> > However if the text changes the test should fail.
> 
> I have to respectfully disagree with you on this one. Some of the suggested changes below get at this same question. Let me try and restate my opinion and see if it changes your mind at all :) If not, perhaps we should take the discussion to others on the team.
> 
> The two primary goals of the Cypress tests are to:
> 
> 1. Watch out for user-facing regressions that we might otherwise not notice
> 2. Test user interactions that are difficult or impossible to test with unit tests
> 
> Testing to see that content that is hard-coded into a template has not changed does not contribute to either of these goals. Hard-coded content will never* change unless those lines of code are edited, and if someone does that, they're always** doing it intentionally and would simply be annoyed by the resulting Cypress test failure.
> 
> _* Unless that hard-coded content is later manipulated by JavaScript. In the case of these interactions, that is not the case. It is simply made visible or hidden._
> _** Unless they made a typo while editing that file. Possible, but unlikely, and hopefully caught in code review._
> 
> The ideal use case for should-contain assertions is something like the cost calculations on Your Financial Path. We want to ensure that given a certain set of input values, the resulting total that is output on the page is calculated consistently. But for the three types of interactions we have on these Choose a Student Loan and Manage Your College Money pages, all we really need to confirm is that the selected content becomes visible (or goes away).
> 
> Let me know what you think!

Perhaps we could discuss the testing goals after a daily scrum with the rest of the team to get this PR merged.
If any lines of code are changed, and tests still pass, there are either missing tests, or a badly written test.

I don't think an assertion that just checks if something `is.visible` or `is.not.visible` when clicked is a good test.
When reading the documentation https://docs.cypress.io/api/commands/should.html
I see a good example of how to use should as follows:
`cy
  .get('nav')                          // yields <nav>
  .should('be.visible')                // yields <nav>
  .should('have.css', 'font-family')   // yields 'sans-serif'
  .and('match', /serif/)               // yields 'sans-serif'`

Instead of using `match` I am using `contain` to verify an answer is visible and contains a string of text.
Each click on a page reveals a different answer.
The tests should verify that the correct answer to a question is being provided.
If an answer to a question get changed unintentionally, then a test will fail which will catch defects before deployment.

File: test/cypress/integration/pages/your-financial-path-to-graduation.js [hunk 21]
```diff
@@ -0,0 +1,219 @@
...
+      page.setText( 'scholarships__state', '17000');
+      page.setText( 'scholarships__school', '18000');
+      page.setText( 'scholarships__other', '19000');
+      cy.get( '[data-financial-item="total_grantsScholarships"]' ).should( 'contain', '$158,500' );
+      page.click( 'Next step' );
+      page.setText( 'workStudy__workStudy', '50000');
+      cy.get( '[data-financial-item="total_workStudy"]' ).should( 'contain', '$50,000' );
+      page.click( 'Next step' );
+      page.setText('loans__directUnsub', '90000');
+      cy.get( '[data-financial-item="total_fedLoans"]' ).should( 'contain', '$5,442' );
```
Comment (Scotchester) [inline test/cypress/integration/pages/your-financial-path-to-graduation.js:107]: I'm now seeing this assertion fail here and in the bachelor's degree section. Any idea why that might be?

Comment (Scotchester) [reply test/cypress/integration/pages/your-financial-path-to-graduation.js:107]: I wonder if it has to do with the program type radio buttons? The snapshots I'm seeing in the Cypress interface do not show them getting selected.

**Review by willbarton — CHANGES_REQUESTED** [review]

File: test/cypress/integration/pages/choose-a-student-loan.js [hunk 22]
```diff
@@ -0,0 +1,73 @@
...
+
+describe( 'Paying for College', () => {
+  beforeEach( () => {
+    page.openLoan();
+  } );
+  describe( 'Choose a student loan', () => {
+    it( 'should display answers to each question', () => {
+      cy.get( '#_o1' ).click();
+      cy.get( '#_o1' ).should( 'be.visible' )
+        .and( 'contain', 'I have to borrow money for school. What are my options?' );
```
[the remark this answers is not in the document]
Comment (cwdavies) [reply test/cypress/integration/pages/choose-a-student-loan.js]: Removed checking for text that's hard-coded in a template

File: test/cypress/pages/paying-for-college/your-financial-path-to-graduation.js [hunk 23]
```diff
@@ -0,0 +1,62 @@
...
+    cy.visit( `/paying-for-college2/understanding-your-financial-aid-offer/api/program/${ name }/` );
+  }
+
+  openSchool( id ) {
+    cy.visit( `/paying-for-college2/understanding-your-financial-aid-offer/api/school/${ id }.json` );
+  }
+
+  school( term ) {
+    cy.visit( `/paying-for-college2/understanding-your-financial-aid-offer/api/search-schools.json?q=${ term }` );
+  }
```
[the remark this answers is not in the document]
Comment (cwdavies) [reply test/cypress/pages/paying-for-college/your-financial-path-to-graduation.js]: Remove tests for API responses

File: test/cypress/pages/paying-for-college/your-financial-path-to-graduation.js [hunk 24]
```diff
@@ -0,0 +1,62 @@
+export class PfcFinancialPathToGraduation {
+
+  openConstants() {
+    cy.visit( '/paying-for-college2/understanding-your-financial-aid-offer/api/constants/' );
+  }
+
+  openNationalStats() {
+    cy.visit( '/paying-for-college2/understanding-your-financial-aid-offer/api/national-stats/' );
+  }
```
[the remark this answers is not in the document]
Comment (cwdavies) [reply test/cypress/pages/paying-for-college/your-financial-path-to-graduation.js]: Remove tests for API responses

File: test/cypress/integration/pages/your-financial-path-to-graduation.js (hunk 21 above)
Comment (cwdavies) [reply test/cypress/integration/pages/your-financial-path-to-graduation.js:102]: I cannot reproduce this issue. The program type radio buttons are hidden behind the school search results.
I have made some minor changes to the `your-financial-path-to-graduation` test.
Please attach a screen snapshot of the Cypress interface showing an assertion failure.

Comment (Scotchester) [reply test/cypress/integration/pages/your-financial-path-to-graduation.js:102]: Here's what I'm seeing (this is on the Bachelor's section, but it also happens on the Graduate section in the same spot):

Comment (cwdavies) [reply test/cypress/integration/pages/your-financial-path-to-graduation.js]: I believe this has been resolved by setting loans__directUnsub to 2000.
It seems you cannot have a direct Unsub loan as high as 90000.

**Review by Scotchester — APPROVED** [review]
OK, that fixed 'em! I think we're good to go here.

**Pull request merged**